### PR TITLE
Use keystone V3 API

### PR DESCRIPTION
**Pull request by rhafer**
This depends on: https://github.com/crowbar/barclamp-keystone/pull/280

### CODE REVIEW

**Comment by mjura** [comment]
After keystone change will be merged +1 from me

File: chef/data_bags/crowbar/bc-template-glance.json [hunk 1]
```diff
@@ -66,7 +67,7 @@
     "glance": {
       "crowbar-revision": 0,
       "crowbar-applied": false,
-      "schema-revision": 20,
+      "schema-revision": 21,
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/bc-template-glance.json]: can you add some extra revision space for tex backports? like bump to 30 or so?

**Comment by rhafer** [comment]
@dirkmueller Actually, I guess this is going to be backported to tex. So shouldn't we avoid to have this migration in tex and master with different revisions? How do we usually handle situations like this?

**Comment by dirkmueller** [comment]
@rhafer ah, good point. if its the first migratoin (it is), then we can just reuse the same number for tex and master. in other cases we created a new one from the stable branch and changed the migration in master to only do something if the previous one didn't run yet.

**Comment by dirkmueller** [comment]
Alright then, so +1

File: chef/cookbooks/glance/templates/default/glance-api.conf.erb [hunk 2]
```diff
@@ -483,7 +484,7 @@ filesystem_store_datadir = <%= node[:glance][:filesystem_store_datadir] %>
 
 # Version of the authentication service to use
 # Valid versions are '2' for keystone and '1' for swauth and rackspace
-swift_store_auth_version = 2
+swift_store_auth_version = <%= @keystone_settings['api_version'] %>
```
Comment (vuntz) [inline chef/cookbooks/glance/templates/default/glance-api.conf.erb]: Are you sure this is the API version? The comment above the line seems to say something else.

Comment (rhafer) [inline chef/cookbooks/glance/templates/default/glance-api.conf.erb]: Yes, I am. I even submitted a review for the glance default config upstream :)

**Pull request merged**